programs/ssl/ssl_pthread_server.c

```diff
@@ -457,11 +556,22 @@ int main( void )
     if( ( ret = mbedtls_net_bind( &listen_fd, NULL, "4433", MBEDTLS_NET_PROTO_TCP ) ) != 0 )
     {
         mbedtls_printf( " failed\n  ! mbedtls_net_bind returned %d\n\n", ret );
-        goto exit;
+        goto dealloc;
     }
 
     mbedtls_printf( " ok\n" );
 
+    if( ( ret = mbedtls_net_set_nonblock( &listen_fd ) ) != 0 )
```

Comment:
This is sample code for end users, so I think it would be good to explain in a comment why we're making this call.